### ISSUE TITLE
Snakemake 8.01 --rulegraph error

### ISSUE THREAD

**Issue by niekwit** [snakemake/snakemake#2577]
**Describe the bug**
I normally create a rule graph as follows:
```
$ snakemake --version
7.32.4
$ snakemake --forceall --rulegraph | grep -v '\-> 0\|0\[label = \"all\"' | dot -Tpdf > images/rule_graph.pdf
Building DAG of jobs...
$ 
```
I installed Snakemake 8 in another Mamba env and on the same workflow/data I want to make the rule graph again, but now I get an error:
```
$ snakemake --version
8.0.1
$ snakemake --forceall --rulegraph | grep -v '\-> 0\|0\[label = \"all\"' | dot -Tpdf > images/rule_graph.pdf
Traceback (most recent call last):
  File "/home/niek/mambaforge/envs/snakemake8/lib/python3.12/site-packages/snakemake/cli.py", line 1936, in args_to_api
    dag_api.printrulegraph()
  File "/home/niek/mambaforge/envs/snakemake8/lib/python3.12/site-packages/snakemake/api.py", line 606, in printrulegraph
    self.workflow_api._workflow.printrulegraph()
  File "/home/niek/mambaforge/envs/snakemake8/lib/python3.12/site-packages/snakemake/workflow.py", line 888, in printrulegraph
    ignore_incomplete=self.execution_settings.ignore_incomplete,
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'ignore_incomplete'
$ 
```

**Additional context**
On a different Snakemake workflow with different data I get the exact same error

**Comment by KasperSkytte** [comment]
Duplicate of #2566

**Comment by abonte** [comment]
Solved in v8.4.0
